src/Package/Impl/Repl/Commands/ReplCommandController.cs

```diff
@@ -159,18 +159,17 @@ public class ReplCommandController : ViewController {
         }
 
         private bool HandleCancel(RCompletionController controller) {
-            if (!controller.HasActiveCompletionSession && !controller.HasActiveSignatureSession(TextView)) {
-                // If session is reading user input, do not terminate it
-                if (!Workflow.RSession.IsReadingUserInput) {
-                    Workflow.Operations.CancelAsync().DoNotWait();
-                    // Post interrupt command which knows if it can interrupt R or not
-                    Workflow.Shell.PostCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdInterruptR);
-                    return true;
-                }
-            } else {
+            if (controller.HasActiveCompletionSession || controller.HasActiveSignatureSession(TextView)) {
                 controller.DismissAllSessions();
                 return true;
             }
+
+            // If session is reading user input, do not terminate it
+            if (!Workflow.RSession.IsReadingUserInput && Workflow.RSession.CanInterrupt(Services)) {
```

Comment:
`CanInterrupt` is checked in `InterruptRCommand` already. Why do we need to check it here as well?

Reply:
I guess we can try and post command anyway... Let me see.

Reply:
OK, simplified it back